src/plugins/data/public/ui/query_editor/language_selector.tsx

```diff
@@ -59,51 +61,60 @@ export const QueryLanguageSelector = (props: QueryLanguageSelectorProps) => {
     setPopover(!isPopoverOpen);
   };
 
-  const languageOptions: Array<{ label: string; value: string }> = [];
-
-  languageService.getLanguages().forEach((language) => {
-    if (
-      (language && props.appName && !language.editorSupportedAppNames?.includes(props.appName)) ||
-      languageService.getUserQueryLanguageBlocklist().includes(language?.id) ||
-      (datasetSupportedLanguages && !datasetSupportedLanguages.includes(language.id))
-    )
-      return;
-    languageOptions.unshift(mapExternalLanguageToOptions(language));
-  });
-
-  const selectedLanguage = {
-    label:
-      (languageOptions.find(
-        (option) => (option.value as string).toLowerCase() === currentLanguage.toLowerCase()
-      )?.label as string) ?? languageOptions[0].label,
-  };
+  const languageOptions = useMemo(() => {
+    const options: Array<{ label: string; value: string }> = [];
 
-  const handleLanguageChange = (newLanguage: string) => {
-    setCurrentLanguage(newLanguage);
-    props.onSelectLanguage(newLanguage);
-  };
+    languageService.getLanguages().forEach((language) => {
+      const isSupported =
+        !datasetSupportedLanguages || datasetSupportedLanguages.includes(language.id);
+      const isBlocklisted = languageService.getUserQueryLanguageBlocklist().includes(language?.id);
+      const isAppSupported =
+        !props.appName || language?.editorSupportedAppNames?.includes(props.appName);
+
+      if (!isSupported || isBlocklisted || !isAppSupported) return;
```

Comment:
should this be !isBlocklisted?

Reply:
tbh i added this advanced setting in case host wanted to prevent specific languages from showing but i doubt it will be used in the situation where this was in experimental and a language wasn't working. this should be removed in general.

Reply:
What is that advanced setting supposed to do?